Move /api/v1/sections/valid_scripts strings to dashboard

maddiedierker opened the pull request:
This PR is follow-up for #22507 to move strings consumed by `/api/v1/sections/valid_scripts` to dashboard.

### What it does
- Points the `/api/v1/sections/valid_scripts` endpoint to already-existing script titles in `dashboard/config/locales/`. Some of the script names in dashboard are slightly different than those in the [i18n Google Sheet](https://docs.google.com/spreadsheets/d/1Tq7VqZALgRA0wYk0HDfEOTyRI0TM2Dir2rloXIPGCgU/edit?usp=sharing), so this change was done with approval by the curriculum and product teams.
- Adds the category names for scripts to `dashboard/config/locales/`

### Follow-up Work
- Once nothing depends on the old `/v2/sections/valid_scripts` endpoints and the associated name/category strings in the i18n GSheet, those strings can be removed 🎉

davidsbailey commented:
@Hamms , you've probably thought of this already, but i'd like to understand the impact of this change in other languages. if a string is moved from pegasus to dashboard without its english contents changing, will we still pick up the existing pegasus translations when the new string is requested in another language in dashboard?

davidsbailey commented on dashboard/app/models/script.rb:
```diff
@@ -1087,10 +1087,10 @@ def alternate_script(user)
   def assignable_info
     info = ScriptConstants.assignable_info(self)
-    info[:name] = I18n.t("#{info[:name]}_name", default: info[:name])
+    info[:name] = I18n.t("data.script.name.#{info[:name]}.title", default: info[:name])
```
While it looks like the script tests added in the previous PR may still pass, they will only be testing the "default" case here. It could be worth adding a test that the i18n strings are getting picked up. There is an example of where we do this in pegasus tests: https://github.com/code-dot-org/code-dot-org/blob/310f220465d5290a970e22089d9cf3cde5c224aa/pegasus/test/test_section_api_helpers.rb#L356-L362

I'd recommend at least giving this a quick try.

maddiedierker replied:
great point! i worried about adding a test that relied on a specific script, but i suppose CS Fundamentals won't be going away any time soon 😆

Hamms commented:
I'm not positive how duplicates work in crowdin across files, but I believe that yes, these "new" translations will automatically be flagged as duplicates of the existing ones and will be auto-translated.

Note that updating our system with those translations will require an i18n sync. I could do a special one just for this PR if necessary, or we could just wait til Monday when the next regular sync is scheduled. How much of a bummer would it be for these to be exclusively in English for a couple days?

maddiedierker commented:
@Hamms i think the sync can wait until monday since i haven't updated consumers to use this endpoint yet. 

re: removing the i18n strings, the script names/categories can be deleted once i've removed all dependencies on the old endpoint (`/v2/sections/valid_scripts`) and i've updated this line in `teacher-dashboard/index.haml`:
https://github.com/code-dot-org/code-dot-org/blob/6b1b296b97c4f123c59a9bfc793dbe0cebf1a563/pegasus/sites.v3/code.org/public/teacher-dashboard/index.haml#L41

maddiedierker commented:
sorry for the additional push after approval--i double-checked the i18n strings and found a category i missed (`full_courses_category_name`) 😱 and decided to alphabetize the category strings while i was there
